includes/core/classes/class-event-feed.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * Handles event feeds for GatherPress.
+ *
+ * This class is responsible for improving the default RSS feeds for events,
+ * including filtering to show only upcoming events, ordering by start date,
+ * and customizing excerpts to show event details.
+ *
+ * @package GatherPress\Core
+ * @since 1.0.0
+ */
+
+namespace GatherPress\Core;
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit; // @codeCoverageIgnore
+
+use GatherPress\Core\Traits\Singleton;
+use WP_Query;
+
+/**
+ * Class Event_Feed.
+ *
+ * Manages event feeds for GatherPress.
+ *
+ * @since 1.0.0
+ */
+class Event_Feed {
+	/**
+	 * Enforces a single instance of this class.
+	 */
+	use Singleton;
+
+	/**
+	 * Class constructor.
+	 *
+	 * This method initializes the object and sets up necessary hooks.
+	 *
+	 * @since 1.0.0
+	 */
+	protected function __construct() {
+		$this->setup_hooks();
+	}
+
+	/**
+	 * Set up hooks for various purposes.
+	 *
+	 * This method adds hooks for different purposes as needed.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	protected function setup_hooks(): void {
+		// Customize RSS excerpts for events.
+		add_filter( 'the_excerpt_rss', array( $this, 'customize_event_excerpt' ) );
+		add_filter( 'the_content_feed', array( $this, 'customize_event_content' ) );
```

Comment:
I'm not really happy with that hard replacements at all and would like to keep some more flexibility for the editor and or theme designer.

I haven't tested this, but I'd like to try hooking into [`get_template_part_slug`](https://developer.wordpress.org/reference/hooks/get_template_part_slug/) or later and provide a default template from within GatherPress. This template could be overwrittren by a theme or within the block-editor or both.

Reply:
Maybe even better would be using register_block_template() which allows to hook into the default template hierachy, just by setting the $template_name correctly to match a default template.

In our case this should be "gatherpress/feed_gatherpress_event" I think.

Using this, GatherPress could provide a default different from wp core and allow theme designers to adjust the template (even inside the block editor).

Reply:
@carstingaxion OK, I implemented some filters to make sure it can be overwritten if need be:
https://github.com/GatherPress/gatherpress/pull/1146/commits/0608028e551016bd941b9be28008d496bb153328